### ISSUE TITLE
Got error while compiling.

### ISSUE THREAD

**Issue by shankhadevpadam** [cretueusebiu/laravel-vue-spa#80]
I got this error. I have using vue 2.5.16 and vue-template-compiler 2.5.16
```
[Vue warn]: Failed to mount component: template or render function not defined.

found in

---> <Anonymous>
       <Root>
```

**Comment by kwame101** [comment]
Your question is very vague. The reason why vue is warning you is because it can't find the root id in the html document, so what you need to do is the mount method e.g 
`import Index from '../Index.vue;`
`new Vue(Index).**mount**('#root');`

I'm sure this project already have these things pre done, just follow the examples in **pages** folder

**Comment by cretueusebiu** [comment]
Try yarn, so it can use the tested and locked versions of all packages. 
If you want to use the latest of everything you'll have to do some testing on your end.

**Comment by shankhadevpadam** [comment]
After release of ``` vue-loader v13.0.0 ``` we need to add default in ``` require ``` so i have fixed by adding the default in 
```
import Loading from './Loading'

// Load layout components dynamically.
const requireContext = require.context('~/layouts', false, /.*\.vue$/)

const layouts = requireContext.keys()
  .map(file =>
    [file.replace(/(^.\/)|(\.vue$)/g, ''), requireContext(file)]
  )
  .reduce((components, [name, component]) => {
    components[name] = component.default
    return components
  }, {})
```

**Comment by cretueusebiu** [comment]
Fixed with 4.1, always uses `esModule: true`
